<?xml version="1.0"?>
<launch>
    <arg name="eye_on_hand" doc="eye-on-hand instead of eye-on-base" default="false"/>
    <arg name="namespace_prefix" default="kinova_gen3_lite" />
    <!-- <arg name="tracking_marker_frame" default="aruco_marker_frame" />
    <arg name="robot_effector_frame" default="tool_frame" /> -->


    <arg if="$(arg eye_on_hand)" name="namespace" value="$(arg namespace_prefix)_eye_on_hand" />
    <arg unless="$(arg eye_on_hand)" name="namespace" value="$(arg namespace_prefix)_eye_on_base" />

    <!--it is possible to override the link names saved in the yaml file in case of name clashes, for example-->
    <arg if="$(arg eye_on_hand)" name="robot_effector_frame" default="tool_frame" />
    <arg unless="$(arg eye_on_hand)" name="robot_base_frame" default="base_link" />
    <arg name="tracking_base_frame" default="" />
    
    <arg name="inverse" default="false" />
    <arg name="calibration_file" default="$(find first_demo)/config/$(arg namespace).yaml" />
    
    <!--publish hand-eye calibration-->
    <group ns="$(arg namespace)">
        <param name="eye_on_hand" value="$(arg eye_on_hand)" />
        <param unless="$(arg eye_on_hand)" name="robot_base_frame" value="$(arg robot_base_frame)" />
        <param if="$(arg eye_on_hand)" name="robot_effector_frame" value="$(arg robot_effector_frame)" />
        <param name="tracking_base_frame" value="$(arg tracking_base_frame)" />
        <param name="inverse" value="$(arg inverse)" />
        <param name="calibration_file" value="$(arg calibration_file)" />
        <node name="$(anon handeye_publisher)" pkg="easy_handeye" type="publish.py" output="screen"/>
    </group>

    <!-- <node pkg="tf" type="static_transform_publisher" name="camera_calibration_tf" 
    args="0 0 0 0 -1.5707963268 1.5707963268 camera_link camera_link_turned 100" /> -->
  
  
     <!-- evaluate the accuracy of the hand-eye calibration  -->
    <!-- <node if="$(arg evaluation)" ns="$(arg namespace)" name="calibration_evaluator" pkg="rqt_easy_handeye" type="rqt_calibration_evaluator" respawn="false" output="screen" >
        <param name="eye_on_hand" value="$(arg eye_on_hand)" />
        <param name="robot_base_frame" value="$(arg robot_base_frame)" />
        <param name="robot_effector_frame" value="$(arg robot_effector_frame)" />
        <param name="tracking_base_frame" value="$(arg tracking_base_frame)" />
        <param name="tracking_marker_frame" value="$(arg tracking_marker_frame)" />
    </node> -->

</launch>
